### ISSUE TITLE
Scaling Term is multiplied twice on calculating Focal Loss

### ISSUE THREAD

**Issue by bilzard** [ultralytics/yolov5#6517]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Training

### Bug

Both BCELoss[1] and FocalLoss[2] have a weight factor.
In `BCELoss`, we can control it by `pos_weight` parameter, whereas in `FocalLoss`, we can control it by `alpha`.
This means that in the YOLOv5 code base, weight factor is computed twice.
This is a mistake.

Reference:

* [1] BCEWithLogitsLoss: https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html
* [2] FocalLoss: https://arxiv.org/pdf/1708.02002.pdf

### Environment

- YOLO: YOLOv5 (latest; how to check the yolo version?)
- CUDA: 11.6 (Tesla T4, 15360MiB)
- OS: Ubuntu 18.04.6 LTS (Bionic Beaver)
- Python: 3.8.12

### Minimal Reproducible Example

BCE loss and FocalLoss(gamma=0.0) are different value. They should be equal.

```console
$ python test.py                                                                                                                                                   (pytorch_x86)
...
assert FocalLoss_before(BCE(pw=1), gamma=0) != BCE(pw=1)
ok
```
[test.py.zip](https://github.com/ultralytics/yolov5/files/7990662/test.py.zip)

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@bilzard yes, they should be equal value like you say. I'll take a look at your PR, thank you!
